lib/kadalulib.py

```diff
@@ -77,6 +77,34 @@ def is_gluster_mount_proc_running(volname, mountpoint):
         return proc.returncode == 0
 
 
+def is_host_reachable(hosts, port):
+    """Check if glusterd is reachable in the given node"""
+    timeout = 5
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    sock.settimeout(timeout)
+    for host in hosts:
+        try:
+            sock.connect((host, int(port)))
+            sock.shutdown(socket.SHUT_RDWR)
+            return True
+        except socket.error as msg:
+            logging.error(logf("Failed to open socket connection",
+                               error=msg, host=host))
+            continue
+        finally:
+            sock.close()
+    return False
+
+
+def reachable_hosts(hosts):
```

Comment:
nit: Function name is plural, may confuse others who read code. Change it to `reachable_host`

Reply:
Done